controllers/activemqartemis_controller_deploy_operator_test.go

```diff
@@ -243,10 +263,140 @@ var _ = Describe("artemis controller", Label("do"), func() {
+												"args":    []interface{}{"-c", dummyPythonScript},
+												"image":   "registry.access.redhat.com/ubi9/python-312-minimal",
+											},
+										},
+									},
+								},
+							},
+						},
+					}
+
+					By("deploy dummy broker with cr: " + crName)
+					cr := brokerv1beta1.ActiveMQArtemis{
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "ActiveMQArtemis",
+							APIVersion: brokerv1beta1.GroupVersion.Identifier(),
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      crName,
+							Namespace: tempNs,
+						},
+						Spec: brokerv1beta1.ActiveMQArtemisSpec{
+							DeploymentPlan: brokerv1beta1.DeploymentPlanType{
+								ReadinessProbe: &corev1.Probe{
+									ProbeHandler: corev1.ProbeHandler{
+										TCPSocket: &corev1.TCPSocketAction{
+											Port: intstr.IntOrString{
+												IntVal: 8161,
+											},
+										},
+									},
+								},
+							},
+							ResourceTemplates: []brokerv1beta1.ResourceTemplate{
+								{
+									Selector: &brokerv1beta1.ResourceSelector{Kind: &ssKind},
+									Patch:    statefulSetPatch,
+								},
+							},
+						},
+					}
+
+					Expect(k8sClient.Create(ctx, &cr)).Should(Succeed())
+				}
+
+				createdSs := &appsv1.StatefulSet{}
+				createdCr := &brokerv1beta1.ActiveMQArtemis{}
+
+				for i := 0; i < count; i++ {
+					crName := crNamePrefix + "-" + strconv.Itoa(i)
+
+					By("checking dummy broker with cr: " + crName)
+					ssKey := types.NamespacedName{Name: namer.CrToSS(crName), Namespace: tempNs}
+					Eventually(func(g Gomega) {
+						g.Expect(k8sClient.Get(ctx, ssKey, createdSs)).Should(Succeed())
+					}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+					crKey := types.NamespacedName{Name: crName, Namespace: tempNs}
+					Eventually(func(g Gomega) {
+						g.Expect(k8sClient.Get(ctx, crKey, createdCr)).Should(Succeed())
+						g.Expect(meta.IsStatusConditionTrue(createdCr.Status.Conditions, brokerv1beta1.ValidConditionType)).Should(BeTrue())
+					}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+				}
+
+				podList := &corev1.PodList{}
+				Expect(k8sClient.List(context.Background(), podList, client.InNamespace(tempNs),
+					client.MatchingLabels{"control-plane": "controller-manager", "name": oprName})).To(Succeed())
+				Expect(len(podList.Items) > 0).To(BeTrue())
+
+				By("checking manager container memory")
+				Eventually(func(g Gomega) {
+					currentMemoryText, err := RunCommandInPodWithNamespace(
+						podList.Items[0].Name, tempNs, "manager",
+						[]string{"cat", "/sys/fs/cgroup/memory.current"})
+					g.Expect(err).To(BeNil())
+
+					currentMemoryBytes, err := strconv.ParseInt(
+						strings.TrimSpace(*currentMemoryText), 10, 64)
+					g.Expect(err).To(BeNil())
+
+					g.Expect(currentMemoryBytes < 256*1024*1024).Should(BeTrue())
+				}, timeout, interval).Should(Succeed())
+
+				uninstallOperator(false, tempNs)
+				deleteNamespace(tempNs, false, Default)
```

Comment:
Another question, why don't we wait for the deletion of the namespaces here? Is it because a namespace with a 1000 brokers is very slow to be deleted?
I assume that the fact that this namespace isn't deleted in time for the next test isn't impacting the next test, otherwise it would fail. Therefore, if the presence of this namespaces isn't colliding with other tests, maybe we should simply leave it there?